testcontainers-dapr/src/main/java/io/dapr/testcontainers/QuotedBoolean.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2024 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.testcontainers;
+
+public class QuotedBoolean {
```

Comment:
What is this for?

Reply:
This was needed because the Dapr Statestore Component has a metadata property that receives a boolean, but it should be quoted as a String with double quotes: 

```
metadata:
  name: statestore
apiVersion: dapr.io/v1alpha1
kind: Component
spec:
  metadata:
  - name: actorStateStore
    value: "true"
  type: state.in-memory
  version: v1
```

If you use a string directly, this is marshalled to YAML with single quotes:
```
"actorStateStore", "true"
```

```
metadata:
  name: statestore
apiVersion: dapr.io/v1alpha1
kind: Component
spec:
  metadata:
  - name: actorStateStore
    value: 'true'
  type: state.in-memory
  version: v1
```

This breaks, so the marshaller needs that helper class to understand what to do with that special string that needs double quotes.

Reply:
This seems to be an implementation detail, the user should not need to be exposed to this in the interface. It means this class should not be public, which might trigger some other design decisions to change too.

Reply:
all removed